src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java

```diff
@@ -0,0 +1,2700 @@
+         * 
+         * @throws Exception
+         */
+        @Test
+        public void testStatementOutParamGetsTwiceInOut() throws Exception {
+            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
+            Connection con = DriverManager.getConnection(connectionString);
+            Statement stmt = con.createStatement();
+            stmt.executeUpdate(
+                    "if exists (select * from dbo.sysobjects where id = object_id(N'[dbo].[sp_ouputP]') and OBJECTPROPERTY(id, N'IsProcedure') = 1) DROP PROCEDURE [sp_ouputP]");
+            stmt.executeUpdate(
+                    "CREATE PROCEDURE [sp_ouputP] ( @p2_smallint smallint,  @p3_smallint_out smallint OUTPUT) AS SELECT @p3_smallint_out=@p3_smallint_out +1 RETURN @p2_smallint + 1");
+
+            CallableStatement cstmt = con.prepareCall("{  ? = CALL [sp_ouputP] (?,?)}");
+            cstmt.registerOutParameter(1, Types.INTEGER);
+            cstmt.setObject(2, Short.valueOf("1"), Types.SMALLINT);
+            cstmt.setObject(3, Short.valueOf("100"), Types.SMALLINT);
+            cstmt.registerOutParameter(3, Types.SMALLINT);
+            cstmt.execute();
+            assertEquals(cstmt.getInt(1), 2, "Wrong value");
+            assertEquals(cstmt.getInt(3), 101, "Wrong value");
+
+            cstmt.setObject(2, Short.valueOf("10"), Types.SMALLINT);
+            cstmt.execute();
+            assertEquals(cstmt.getInt(1), 11, "Wrong value");
+            assertEquals(cstmt.getInt(3), 101, "Wrong value");
+
+        }
+
+        /**
+         * Tests resultset parameters
+         * 
+         * @throws Exception
+         */
+        @Test
+        public void testResultSetParams() throws Exception {
+            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
+            Connection conn = DriverManager.getConnection(connectionString);
+            Statement stmt = conn.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_UPDATABLE);
+
+            try {
+                stmt.executeUpdate("drop table if exists " + tableName);
+            }
+            catch (Exception ex) {
+            }
+            ;
+            try {
+                stmt.executeUpdate("IF EXISTS (select * from sysobjects where id = object_id(N'" + procName
+                        + "') and OBJECTPROPERTY(id, N'IsProcedure') = 1)" + " DROP PROCEDURE " + procName);
+            }
+            catch (Exception ex) {
+            }
+            ;
+            stmt.executeUpdate("create table " + tableName + " (col1 int, col2 text, col3 int identity(1,1) primary key)");
+            stmt.executeUpdate("Insert into " + tableName + " values(0, 'hello')");
+            stmt.executeUpdate("Insert into " + tableName + " values(0, 'yo')");
+            String query = "create procedure " + procName + " @col1Value int, @col2Value varchar(512) OUTPUT AS BEGIN SELECT * from " + tableName
+                    + " where col1=@col1Value SET @col2Value='wassup' END";
```

Comment:
do not use strings 'yo' and 'wassap'. may need to retest to make sure nothing broke once strings are modified